### ISSUE TITLE
[Bug]: Building @zlib generates libzlib.so instead of libz.so

### ISSUE THREAD

**Issue by tzm-luminar** [bazelbuild/bazel-central-registry#2226]
### What happened?

I added `bazel_dep(name = "zlib", version = "1.3.1")` to my `MODULE.bazel` then ran `bazel build @zlib`.
The output was:
```
INFO: Analyzed target @@zlib~//:zlib (109 packages loaded, 1163 targets configured).
INFO: Found 1 target...
Target @@zlib~//:zlib up-to-date:
  bazel-bin/external/zlib~/libzlib.a
  bazel-bin/external/zlib~/libzlib.so
```

I expected the generated files to be libz.[a|so], not libzlib.[a|so]

### Version

Development (host) and target OS/architectures: Linux

Output of `bazel --version`: bazel 7.1.2

Version of relevant rules from the `WORKSPACE` or `MODULE.bazel` file: ?

Language(s) and/or frameworks involved: ?

### How to reproduce

```shell
Try to build zlib from Bazel Central Registry
```

**Comment by dmitry-j-mikhin** [comment]
A similar issue has been fixed for version 1.2.13.bcr.1 - https://github.com/bazelbuild/bazel-central-registry/pull/717.
But this fix is ​​missing in version 1.3.1. I'll try to do PR.

**Comment by dmitry-j-mikhin** [comment]
PR - https://github.com/bazelbuild/bazel-central-registry/pull/2480
